tests/FSharp.Compiler.ComponentTests/Language/ModuleAsTypeTests.fs

```diff
@@ -0,0 +1,73 @@
+﻿module FSharp.Compiler.ComponentTests.Language.ModuleAsTypeTests
+
+open Xunit
+open FSharp.Test.Compiler
+
+[<Fact>]
+let ``Module can be used in typeof in lang preview`` () =
+    Fsx """
+let actual = typeof<module FSharp.Core.LanguagePrimitives>.Name
```

Comment:
```suggestion
[<Theory>]
[<InlineData("typeof")>]
[<InlineData("typedefof")>]
let ``Module can be used in typeof in lang preview`` (typeofKeyword:string) =
    Fsx $"""
let actual = {typeofKeyword}<module FSharp.Core.LanguagePrimitives>.Name
```


I assume typedefof gets automatically supported by your addition as well, can you please test it as well?
i.e. the same test, just using `typedefof` instead of `typeof`.